src/cascadia/TerminalSettingsModel/ActionArgs.idl

```diff
@@ -122,10 +122,22 @@ namespace Microsoft.Terminal.Settings.Model
         All = 0xffffffff,
     };
 
-    [default_interface] runtimeclass NewTerminalArgs {
+    interface INewContentArgs {
+        String Type;
+        Boolean Equals(INewContentArgs other);
+        UInt64 Hash();
+        INewContentArgs Copy();
+        String GenerateName();
+    };
+
+    [default_interface] runtimeclass GenericContentArgs : INewContentArgs{
+        GenericContentArgs();
+        GenericContentArgs(String type);
+    };

```

Comment:
I don't believe this would work well for an extension model as plugins may want arbitrary settings. It may be better to not have a string `Type` and instead have concrete runtimeclasses for each specific builtin Pane type. Extensions may define their own concrete types which we'll then somehow pass through to the plugin to get our `IPaneContent` or whatever. We can also skip that part of course and let such users create and pass `IPaneContent`s directly to the term control constructor(s). In both cases however I would prefer concrete runtimeclasses.

Reply:
Absolutely. As of this commit, the only _real_ `INewContentArgs` is the `NewTerminalArgs`. The settings pane and scratch pane don't really have settings, so in the name of... laziness I guess... I used the smallest possible placeholder for their internal representation. `GenericContentArgs` was never supposed to be anything more than an internal helper. 

In the "near" future where I plan to ship that markdown pane, that'll actually have it's own `MarkdownPaneArgs` with like, a `filePath` property or something.

Reply:
FWIW...

> never .. more than an internal helper

Internal helpers can use `winrt::implements<>` so they don't go in idl files/on the public API! If you need to `try_as` it, that means our `IWhatever` is an insufficient abstraction.

Reply:
"internal" in the sense of "**we** can use this but I'm not really expecting others to use it"

Right now it's also used by `TerminalApp` to also return the type of content for a ScratchpadPane. 

I could theoretically just rename it like, `BaseContentArgs`. Trying to make this entirely internal to TSM results in me basically just creating the same exact internal (non-projected) class up in `TerminalApp`